### PR TITLE
Proof Of Chaos

### PR DESCRIPTION
**Pull request by aadarsh012**

### CODE REVIEW

File: front-end/src/components/Post/GovernanceSideBar/Referenda/ProofOfChaos/QuizForm.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import React, { useContext, useEffect,useState } from 'react';
+import { ApiContext } from 'src/context/ApiContext';
+import { useGetAllAccounts } from 'src/hooks';
+
+import { SubmitQuizAnswers } from './data/quiz-service';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const QuizForm = ({ className, quiz, referendumId, setQuizLevel }: { className?: string, referendumId: Number | null | undefined, quiz: any, setQuizLevel: (level: Number) => void }) => {
+
+	const { accounts, accountsMap, signersMap } = useGetAllAccounts();
```
Comment (alphainfinitus) [inline front-end/src/components/Post/GovernanceSideBar/Referenda/ProofOfChaos/QuizForm.tsx:16]: Please implement the no accounts found, and no Extension etc states.

**Comment by alphainfinitus** [comment]
Need a loading state between the quiz form and voting form.

File: front-end/src/components/Post/GovernanceSideBar/Referenda/ProofOfChaos/QuizForm.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+							submittedOn: Date.now()
+						}
+					}
+				}));
+			}
+		);
+	};
+
+	function onChangeInputs( e: any, questionIndex: string, options: number ) {
+		console.log(e.target.value);
```
Comment (alphainfinitus) [inline front-end/src/components/Post/GovernanceSideBar/Referenda/ProofOfChaos/QuizForm.tsx:73]: remove console logs please

File: front-end/src/components/Post/GovernanceSideBar/Referenda/VoteReferendum.tsx [hunk 3]
```diff
@@ -50,6 +58,77 @@ const VoteReferendum = ({ className, referendumId, address, accounts, onAccountC
...
+                        quizId
+                        referendumIndex
+                        timestamp
+                        version
+                        wallet
+                      }
+                    }
+                  }`
+			}),
+			headers: {
```
Comment (alphainfinitus) [inline front-end/src/components/Post/GovernanceSideBar/Referenda/VoteReferendum.tsx:108]: you can directly pass the subsquidHeaders from the globals

**Comment by aadarsh012** [comment]
Screenshot (382)
Screenshot (384)

**Pull request merged**